front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/adjustConfWithTrainToModify.ts

```diff
@@ -105,3 +111,61 @@ export default function adjustConfWithTrainToModify(
     );
   }
 }
+
+export function adjustConfWithTrainToModifyV2(
+  trainSchedule: TrainScheduleResult,
+  pathSteps: PathStep[],
+  rollingStockId: number,
+  dispatch: Dispatch,
+  usingElectricalProfiles: boolean,
+  osrdActions: ConfSliceActions
+) {
+  const {
+    updateRollingStockID,
+    toggleUsingElectricalProfiles,
+    updateSpeedLimitByTag,
+    updateLabels,
+    updateName,
+    updateDepartureTime,
+    updateInitialSpeed,
+    updatePowerRestrictionRanges,
+    updatePathSteps,
+  } = osrdActions;
+
+  const {
+    options,
+    speed_limit_tag,
+    labels,
+    train_name,
+    start_time,
+    initial_speed,
+    power_restrictions,
+  } = trainSchedule;
+
+  dispatch(updateRollingStockID(rollingStockId));
+  dispatch(updateName(train_name));
+  dispatch(updateDepartureTime(start_time));
+  dispatch(updatePathSteps(pathSteps));
+
+  if (options?.use_electrical_profiles === usingElectricalProfiles)
+    dispatch(toggleUsingElectricalProfiles());
+
+  if (speed_limit_tag) dispatch(updateSpeedLimitByTag(speed_limit_tag));
+
+  if (labels) dispatch(updateLabels(labels));
+
+  if (initial_speed) dispatch(updateInitialSpeed(msToKmh(initial_speed) || 0));
```

Comment:
I think we should always dispatch the initial_speed
```suggestion
  dispatch(updateInitialSpeed(msToKmh(initial_speed) || 0));
```

Reply:
Done (msToKmh can't handle undefined so I added a condition)